packages/react/src/hooks/actions/useDataStoreUpdate.ts

```diff
@@ -0,0 +1,71 @@
+import type {
+  ModelInit,
+  PersistentModel,
+  PersistentModelConstructor,
+} from '@aws-amplify/datastore';
+import { DataStore, Hub } from 'aws-amplify';
+
+import {
+  ACTION_DATASTORE_UPDATE_FINISHED_MESSAGE,
+  ACTION_DATASTORE_UPDATE_FINISHED,
+  ACTION_DATASTORE_UPDATE_STARTED_MESSAGE,
+  ACTION_DATASTORE_UPDATE_STARTED,
+  ACTIONS_CHANNEL,
+  DATASTORE_QUERY_BY_ID_ERROR,
+} from './constants';
+import { getErrorMessage } from '../../helpers/utils';
+
+export interface UseDataStoreUpdateActionOptions<
+  Model extends PersistentModel
+> {
+  model: PersistentModelConstructor<Model>;
+  id: string;
+  fields: UseDataStoreUpdateActionFields<Model>;
+}
+
+type UseDataStoreUpdateActionFields<Model> = ModelInit<
+  Model,
+  { readOnlyFields: 'createdAt' | 'updatedAt' }
+>;
```

Comment:
```suggestion
export interface UseDataStoreUpdateActionOptions<
  Model extends PersistentModel
> {
  model: PersistentModelConstructor<Model>;
  id: string;
  fields: ModelInit<Model, { readOnlyFields: 'createdAt' | 'updatedAt' }>;
}
```

Reply:
Good point, these can definitely be merged. Will update

Reply:
Fixed